catalog/plugin/catalog-plugin-versioning/src/main/java/ddf/catalog/metacard/versioning/HistorianBouncerPolicyPlugin.java

```diff
@@ -0,0 +1,122 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * </p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.metacard.versioning;
+
+import java.io.Serializable;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Predicate;
+import java.util.function.Supplier;
+
+import org.apache.shiro.SecurityUtils;
+import org.apache.shiro.subject.Subject;
+
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.Result;
+import ddf.catalog.data.impl.HistoryMetacardImpl;
+import ddf.catalog.operation.Query;
+import ddf.catalog.operation.ResourceRequest;
+import ddf.catalog.operation.ResourceResponse;
+import ddf.catalog.plugin.PolicyPlugin;
+import ddf.catalog.plugin.PolicyResponse;
+import ddf.catalog.plugin.StopProcessingException;
+import ddf.catalog.plugin.impl.PolicyResponseImpl;
+
+public class HistorianBouncerPolicyPlugin implements PolicyPlugin {
+    private final Predicate<Metacard> isMetacardHistory = (tags) -> tags.getTags()
+            .contains(HistoryMetacardImpl.HISTORY_TAG);
+
+    private final Predicate<Subject> isAdminSubject = (subject) -> SecurityUtils.getSubject()
+            .hasRole("codice-history");
```

Comment:
I know we talked about this already, but just throwing this on here for visibility. This is not the correct way to check if a subject has a particular role. We should always be relying on the PDP for a decision on whether or not a Subject is allowed to do something. This plugin should be attaching this role in the PolicyResponse object. That allows the CatalogFramework to build up a policy for the metacard and then an AccessPlugin will handle whether or not the Subject can imply the role permission.